spec/type.dd

```diff
@@ -344,6 +344,34 @@ ulong  u4 = long(-1); // ok, -1 can be represented in a ulong
     point types.
     )
 
+$(H3 $(LNAME2 vrp, Value Range Propagation))
+
+    $(P Besides type-based implicit conversions, D allows certain integer
+        expressions to implicitly convert to a narrower type after
+        integer promotion. This works by analysing the minimum and
+        maximum possible range of values for each expression.
```

Comment:
```suggestion
        integer promotion. This works by determining the possible range of values of an expression
```

Reply:
possibly link to things like this (https://en.wikipedia.org/wiki/Value_range_analysis) although the form that the compiler performs is more limited.

Reply:
Now added that link, and updated the docs.